evm_arithmetization/src/memory_continuation/memory_continuation_stark.rs

```diff
@@ -53,37 +58,40 @@ pub(crate) fn ctl_data_memory<F: Field>() -> Vec<Column<F>> {
     res
 }
 
-/// CTL filter for memory operations.
-pub(crate) fn ctl_filter<F: Field>() -> Filter<F> {
-    Filter::new_simple(Column::single(FILTER))
+/// Convert `mem_before_values` to a vector of memory trace rows
+pub(crate) fn mem_before_values_to_rows<F: Field>(
+    mem_before_values: &MemBeforeValues,
+) -> Vec<Vec<F>> {
+    mem_before_values
+        .iter()
+        .map(|mem_data| {
+            let mut row = vec![F::ZERO; NUM_COLUMNS];
+            row[FILTER] = F::ONE;
+            row[ADDR_CONTEXT] = F::from_canonical_usize(mem_data.0.context);
+            row[ADDR_SEGMENT] = F::from_canonical_usize(mem_data.0.segment);
+            row[ADDR_VIRTUAL] = F::from_canonical_usize(mem_data.0.virt);
+            for j in 0..VALUE_LIMBS {
+                row[j + 4] = F::from_canonical_u32((mem_data.1 >> (j * 32)).low_u32());
+            }
+            row
+        })
+        .collect()
 }
 
-/// Structure representing the `MemBefore` STARK.
+/// Structure representing the `ContinuationMemory` STARK.
 #[derive(Copy, Clone, Default)]
-pub(crate) struct MemBeforeStark<F, const D: usize> {
+pub(crate) struct MemoryContinuationStark<F, const D: usize> {
     f: PhantomData<F>,
 }
 
-impl<F: RichField + Extendable<D>, const D: usize> MemBeforeStark<F, D> {
+impl<F: RichField + Extendable<D>, const D: usize> MemoryContinuationStark<F, D> {
     pub(crate) fn generate_trace(
         &self,
-        mem_before_values: &MemBeforeValues,
+        final_values: Vec<Vec<F>>,
```

Comment:
It looks like it can either be `mem_before_values` or `final_values`.
```suggestion
        propagated_values: Vec<Vec<F>>,
```